scripts/build_shapes.py

```diff
@@ -267,8 +298,10 @@ def eez(countries, geo_crs, country_shapes, EEZ_gpkg, out_logging=False, distanc
     )
 
     ret_df = ret_df.apply(lambda x: make_valid(x))
-    country_shapes = country_shapes.apply(lambda x: make_valid(x))
+    # country_shapes may consist of different geometries which need to be united
+    country_shapes = country_shapes.apply(lambda x: make_valid(x)).unary_union
```

Comment:
why unary_union?

Reply:
to avoid having multiple offshore shapes which is the case if the onshore dataframe for a country contains multiple geometries

Reply:
My question was because the unary_union merges all shapes of all countries and I was wondering if that introduces (a) additional computational time that may not be required and (b) alter the results as a single eez of one country would be compared to a merged shape by the unary_union.
Have you tested it with multiple countries and check that the output shapes are ok?

Reply:
Yes, I have tested it and it looks ok. But I'd be happy to understand the underlying behaviour of `difference` and `unary_union` better. It looks like there is some implicit grouping by country name and looping inside it. Regarding performance, would be very interested in your opinion. This aspect I haven't considered properly